src/async/parallel.ts

```diff
@@ -1,53 +1,105 @@
-import { AggregateError, flat, fork, list, sort, tryit } from 'radashi'
+import {
+  AggregateError,
+  flat,
+  fork,
+  isNumber,
+  list,
+  sort,
+  tryit,
+} from 'radashi'
 
+type AbortSignal = {
+  readonly aborted: boolean
+  addEventListener(type: 'abort', listener: () => void): void
+  removeEventListener(type: 'abort', listener: () => void): void
+  throwIfAborted(): void
+}
 type WorkItemResult<K> = {
   index: number
   result: K
   error: any
 }
+export type ParallelOptions =
+  | {
+      limit: number
+      signal?: AbortSignal
+    }
+  | number
+
 /**
  * Executes many async functions in parallel. Returns the results from
  * all functions as an array. After all functions have resolved, if
  * any errors were thrown, they are rethrown in an instance of
- * AggregateError.
+ * AggregateError. The operation can be aborted by passing optional AbortSignal,
+ * which will throw an Error if aborted.
  *
  * @see https://radashi.js.org/reference/async/parallel
  * @example
  * ```ts
  * // Process images concurrently, resizing each image to a standard size.
- * const images = await parallel(2, imageFiles, async (file) => {
+ * const abortController = new AbortController();
+ * const images = await parallel(
+ * {
+ *  limit: 2,
+ *  signal: abortController.signal,
+ * },
+ *  imageFiles,
+ *  async file => {
  *   return await resizeImage(file)
  * })
+ *
+ * // To abort the operation:
+ * // abortController.abort()
  * ```
  * @version 12.1.0
  */
 export async function parallel<T, K>(
-  limit: number,
+  options: ParallelOptions,
   array: readonly T[],
   func: (item: T) => Promise<K>,
 ): Promise<K[]> {
   const work = array.map((item, index) => ({
     index,
     item,
   }))
+
+  if (isNumber(options)) {
+    options = {
+      limit: options,
+      signal: undefined,
```

Comment:
No need to define `signal` property here. Although it prevents a prototype lookup, it's a micro-optimization, which we tend to avoid for initial implementations.